### PR TITLE
Add Frostbite status

### PR DESCRIPTION
**Pull request by Bassoonian**
Adds Frostbite as status next to Freeze.

## Description
This PR adds Frostbite as a new status to co-exist with Freeze. A config `B_USE_FROSTBITE` exists to quickly switch Freeze to Frostbite, but moves can be cherry-picked so a combination of both statuses can exist.

## Images
pokeemerald 2023-04-22 18 28 18
pokeemerald 2023-04-22 18 29 23
pokeemerald 2023-04-22 18 29 31

## Task list

- [x] Add status to the game
- [x] Make it function in-battle
- [x] Allow it to be healed with items
- [x] Allow moves to "thaw out"
- [x] Allow enemy moves to "thaw out"
- [x] Allow it to be healed by abilities
- [x] Allow it to be inflicted with relevant configs
- [x] Include other interactions with abilities
- [x] Include other interactions with moves
- [x] Integrate AI interactions with this status
- [x] Battle Pike functionality
- [x] New move descriptions if required

# Tests
?

## Bug/undesirable behaviour list

- [x] `ITEM_EFFECT_CURE_FREEZE` does not discern between strings depending on if it heals a freeze or a frostbite
- [x] There are some palette shenanigans going on when the opponent gets a status

## **Discord contact info**
Jasper#5206

### CODE REVIEW

**Comment by Bassoonian** [comment]
Ready for review! This still needs extensive testing and the `ITEM_CURE_EFFECT_FREEZE` issue still needs to be addressed, but I personally have no clue how. If someone has a solution (we should wait for that before merging it probably), please feel free to PR it!

We'll probably need tests to be written too, but as previously mentioned on Discord, I use OSX and can't do it myself.

File: include/battle_ai_util.h [hunk 1]
```diff
@@ -175,5 +177,6 @@ void IncreaseBurnScore(u8 battlerAtk, u8 battlerdef, u16 move, s16 *score);
 void IncreaseParalyzeScore(u8 battlerAtk, u8 battlerDef, u16 move, s16 *score);
 void IncreaseSleepScore(u8 battlerAtk, u8 battlerDef, u16 move, s16 *score);
 void IncreaseConfusionScore(u8 battlerAtk, u8 battlerDef, u16 move, s16 *score);
+void IncreaseFrostbiteScore(u8 battlerAtk, u8 battlerdef, u16 move, s16 *score);
 
 #endif //GUARD_BATTLE_AI_UTIL_H
```
Comment (CallmeEchoo) [inline include/battle_ai_util.h:182]:
```suggestion
void IncreaseSleepScore(u8 battlerAtk, u8 battlerDef, u16 move, s16 *score);
void IncreaseConfusionScore(u8 battlerAtk, u8 battlerDef, u16 move, s16 *score);
void IncreaseFrostbiteScore(u8 battlerAtk, u8 battlerdef, u16 move, s16 *score);

#endif //GUARD_BATTLE_AI_UTIL_H

```

**Comment by Bassoonian** [comment]
This should now be bugfree and behaviour should be as expected. Further testing is obviously recommended. Written tests would also be welcome if deemed necessary as I can't do them myself (see above).

**Comment by Bassoonian** [comment]
Ready for re-review, now with tests and no more bugs!

**Comment by Jaizu** [comment]
Please give proper credits for fixing the status icons. I know RHH doesn't have a proper credits list yet (it should) but there is no harm into adding the credits into the PR or commit name.

**Comment by Bassoonian** [comment]
I attempted to add you as co-author to the commit, but it doesn't seem to have worked. I also mentioned you on the RHH Discord when I mentioned the fix:
image

**Comment by Jaizu** [comment]
That sounds good enough to me o/

**Comment by floodryan** [comment]
Ive been looking to do this for a while. Is it available to add now? Also would it be hard to merge this with an existing project

**Comment by LOuroboros** [comment]
> Is it available to add now?

It's code, and it's all here, so yes.

If you mean if it's available officially, then the answer is "No." though. As you can see above, this PR is still Open and not Merged.

> Also would it be hard to merge this with an existing project

It depends on whether said project uses the expansion and whether it's kept up-to-date or not.

You're better off pulling the branch that was submitted in this PR and solving the merge conflicts, or alternatively implement the changes manually as they're shown in the Files Changed tab.

Also, for future reference, keep any support questions in the Discord server. The only discussions that should take place here are those that concern the PR itself and the changes it makes with the goal of getting it reviewed, approved and merged.

**Review by ghoulslash — CHANGES_REQUESTED** [review]

File: src/battle_ai_util.c [hunk 2]
```diff
@@ -2892,6 +2892,17 @@ bool32 AI_CanBeBurned(u8 battler, u16 ability)
     return TRUE;
 }
 
+bool32 AI_CanGetFrostbite(u8 battler, u16 ability)
+{
+    if (ability == ABILITY_MAGMA_ARMOR
```
[the remark this answers is not in the document]
Comment (Bassoonian) [reply src/battle_ai_util.c:2897]: I assume that means that `AI_CanBeBurned` is also missing the Comatose check?

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -3208,6 +3211,12 @@ void SetMoveEffect(bool32 primary, u32 certain)
                 gMoveResultFlags |= MOVE_RESULT_DOESNT_AFFECT_FOE;
             }
             break;
+        case STATUS1_FROSTBITE:
+            if (!CanBeFrozen(gEffectBattler))
```
[the remark this answers is not in the document]
Comment (Bassoonian) [reply src/battle_script_commands.c:3215]: Good one! I can't find any information about interactions between sunlight and frostbite, so I assume weather has nothing to do with whether or not a Pokémon can get Frostbite. I'll fix it, thanks!

File: src/battle_ai_util.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (ghoulslash) [reply src/battle_ai_util.c:2897]: Yeah if it missing a comatose check that should be added (may as well on this PR if you have other edits to make)

[the remark this answers is not in the document]
Comment (Bassoonian) [reply src/battle_ai_util.c:2900]: Added the Comatose check to some other status checks too while I was at it :)

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**